docs/7-algorithms-and-data-structures-part-1/hometask/Dijkstra.ts

```diff
@@ -0,0 +1,98 @@
+
+interface DijkstraI {
+    findShortestPath(vertex1: string, vertex2: string): string[];
+    findPointsOfShortestWay(start: string, finish: string, weight: number): string[];
+}
+
+class NodeVertex {
+    nameOfVertex: string;
+    weight: number;
+}
+
+class Vertex {
+    name: string;
+    nodes: NodeVertex[];
+    weight: number;
+
+    constructor(theName: string, theNodes: NodeVertex[], theWeight: number) {
+        this.name = theName;
+        this.nodes = theNodes;
+        this.weight = theWeight;
+    }
+}
+
+export class Dijkstra implements DijkstraI {
+
+    vertices: any;
+    constructor() {
+        this.vertices = {};
+    }
+
+    addVertex(vertex: Vertex): void {
+        this.vertices[vertex.name] = vertex;
+    }
+
+    findPointsOfShortestWay(start: string, finish: string, weight: number): string[] {
+
+        let nextVertex: string = finish;
+        let arrayWithVertex: string[] = [];
+        while (nextVertex !== start) {
+
+            let minWeigth: number = Number.MAX_VALUE;
+            let minVertex: string = "";
+            for (let i of this.vertices[nextVertex].nodes) {
+                if (i.weight + this.vertices[i.nameOfVertex].weight < minWeigth) {
+                    minWeigth = this.vertices[i.nameOfVertex].weight;
+                    minVertex = i.nameOfVertex;
+                }
+            }
+            arrayWithVertex.push(minVertex);
+            nextVertex = minVertex;
+        }
+        return arrayWithVertex;
+    }
+
+
+    findShortestPath(start: string, finish: string): string[] {
+
+        let nodes: any = {};
+        let visitedVertex: string[] = [];
+
+        for (let i in this.vertices) {
+            if (this.vertices[i].name === start) {
+                this.vertices[i].weight = 0;
+
+            } else {
+                this.vertices[i].weight = Number.MAX_VALUE;
+            }
+            nodes[this.vertices[i].name] = this.vertices[i].weight;
+        }
+
+        while (Object.keys(nodes).length !== 0) {
+            let sortedVisitedByWeight: string[] = Object.keys(nodes).sort((a, b) => this.vertices[a].weight - this.vertices[b].weight);

```

Comment:
Whoah, on every iteration you sorting all the visited nodes. The complexity of your solution is astronomy )))